src/parser/content-parser.php

```diff
@@ -61,21 +63,28 @@ public function __construct( $block_registry = null ) {
 	 * Filter out a block from the blocks output based on:
 	 *
 	 * - include parameter, if it is set or
-	 * - exclude parameter, if it is set.
+	 * - exclude parameter, if it is set or
+	 * - whether it is an empty whitespace block
 	 *
 	 * and finally, based on a filter vip_block_data_api__allow_block
 	 *
-	 * @param array  $block Current block.
-	 * @param string $block_name Name of the block.
-	 * @param array  $filter_options Options to be used for filtering, if any.
+	 * @param WP_Block $block Current block.
+	 * @param array    $filter_options Options to be used for filtering, if any.
 	 *
 	 * @return bool true, if the block should be included or false otherwise
 	 *
 	 * @access private
 	 */
-	public function should_block_be_included( $block, $block_name, $filter_options ) {
+	protected function should_block_be_included( WP_Block $block, array $filter_options ) {
+		$block_name        = $block->name;
 		$is_block_included = true;
 
+		// Whitespace blocks are always excluded.
+		$is_whitespace_block = null === $block_name && empty( trim( $block->inner_html ) );
```

Comment:
In reality just about every unit test is testing for this since we are indenting the blocks in our HTML fixtures, which results in whitespace blocks. But I added an explicit test in https://github.com/Automattic/vip-block-data-api/pull/74/commits/9eb3e2cd9f80a72e9b2d8230a6b1918658d2e710

Reply:
Thank you!